### ISSUE TITLE
CommandLine: Error executing cmd. TextEditor  has been disposed.

### ISSUE THREAD

**Issue by jayagami** [VSCodeVim/Vim#5332]
**Describe the bug**

Throwing an exception when excute a simple replace in csv file

```
CommandLine: Error executing cmd=%s/,804_/,189804_/g. err=DISPOSED: TextEditor(vs.editor.ICodeEditor:1,$model6) has been disposed.
```

**To Reproduce**

1. open csv file with vscode (about 6 MB)
2. tab `:` to execute cmd
3. input `%s/,804_/,189804_/g`
4. I can see line are changed, but can not execute other command.
5. close windows, Throwing an exception 

**Expected behavior**

text been replaced.

**Screenshots**

![image](https://user-images.githubusercontent.com/40586590/95042033-21522b80-070b-11eb-926d-b2f286ad9453.png)

**Environment (please complete the following information):**

- Extension (VsCodeVim) version: v1.17.1
- VSCode version: 1.49.3
- OS: Windows

**Additional context**

**Comment by J-Fields** [comment]
I assume you don't have neovim integration enabled (correct me if I'm wrong). About how many substitutions would've been done?

**Comment by jayagami** [comment]
> I assume you don't have neovim integration enabled (correct me if I'm wrong). About how many substitutions would've been done?
> 
> 
> 
> You're throwing a lot at it regardless, but we don't seem to even do substitutions in parallel, so the command is likely still running when you close the editor.

Thanks for your replying.

1. Yes, I don't have neovim

2. About 8000, I will try it again later.

I guess you are right, perhaps the command is still running backwards. 

I can see some line changed in side bar preview, but I can't make sure all line been changed,cause can not do further operation 

Anyway, 8000 is not that much, replace should be very quickly. At least only takes few second in subline or vim

**Comment by J-Fields** [comment]
> Anyway, 8000 is not that much, replace should be very quickly. At least only takes few second in subline or vim

Yeah since we're executing on a single thread that's shared among all extensions, and have to do our updates via the VSCode API, we'll never do as well as sublime/vim, but we can certainly do better. I'll try to get a fix into the next release.
